packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/document_migrator.test.ts

```diff
@@ -847,9 +838,43 @@ describe('DocumentMigrator', () => {
           id: 'cowardly',
           type: 'dog',
           attributes: { name: 'Leslie' },
-          migrationVersion: { dog: '1.0.0' },
           references: [{ id: 'favorite', type: 'toy', name: 'BALL!' }],
-          coreMigrationVersion: kibanaVersion,
+          coreMigrationVersion: '1.0.0',
+          namespace: 'foo-namespace',
+          // there is no 'namespaces' field because no conversion transform was applied; this scenario is contrived for a clean test case but is not indicative of a real-world scenario
+        });
+      });
+
```

Comment:
In the real world there are documents with `coreMigrationVersion: '8.6.0'` even if our code would say this is impossible now, the latest coreMigrationVersion is 8.0.0. I think it would be worth testing that the document migrator doesn't throw when encountering a document with a newer coreMigrationVersion than the latest known coreMigrationVersion like `4.0.0`.

Reply:
There was a similar test case, but that was skipping the `coreMigrationVersion` check. So I added that in 903f7477acc2fc5d741a646ff29c977e1fe5d9a4. Thanks!